core/ir/aarch64/instr_create_api.h

```diff
@@ -658,13 +658,19 @@
     instr_create_0dst_3src((dc), OP_tbnz, (pc), (reg), (imm))
 #define INSTR_CREATE_cmp(dc, rn, rm_or_imm) \
     INSTR_CREATE_subs(dc, OPND_CREATE_ZR(rn), rn, rm_or_imm)
-#define INSTR_CREATE_eor(dc, d, s)                                      \
-    INSTR_CREATE_eor_shift(dc, d, d, s, OPND_CREATE_INT8(DR_SHIFT_LSL), \
-                           OPND_CREATE_INT8(0))
+#define INSTR_CREATE_eor(dc, d, s_or_imm)                                            \
+    opnd_is_immed(s_or_imm)                                                          \
+        ? instr_create_1dst_2src(dc, OP_eor, d, d, s_or_imm)                         \
+        : INSTR_CREATE_eor_shift(dc, d, d, s_or_imm, OPND_CREATE_INT8(DR_SHIFT_LSL), \
+                                 OPND_CREATE_INT8(0))
 #define INSTR_CREATE_eor_shift(dc, rd, rn, rm, sht, sha)                             \
     instr_create_1dst_4src(dc, OP_eor, rd, rn,                                       \
                            opnd_create_reg_ex(opnd_get_reg(rm), 0, DR_OPND_SHIFTED), \
                            opnd_add_flags(sht, DR_OPND_IS_SHIFT), sha)
+#define INSTR_CREATE_csinc(dc, rd, rn, rm, cond) \
+    instr_create_1dst_3src(dc, OP_csinc, rd, rn, rm, cond)
+#define INSTR_CREATE_ubfm(dc, rd, rn, lsb, width) \
```

Comment:
Please replace the `instr_create_1dst_3src(dc, OP_ubfm...` in `XINST_CREATE_slr_s()` with this `INSTR_CREATE_ubfm()`.

Reply:
Thanks, done